packages/frontend-plugin-api/api-report.md

```diff
@@ -157,8 +200,29 @@ export const coreExtensionData: {
   routeRef: ConfigurableExtensionDataRef<RouteRef<AnyRouteRefParams>, {}>;
   navTarget: ConfigurableExtensionDataRef<NavTarget, {}>;
   theme: ConfigurableExtensionDataRef<AppTheme, {}>;
+  component: ConfigurableExtensionDataRef<
+    {
+      ref: ComponentRef<ComponentType<any>>;
```

Comment:
Should the component be generic instead of any? But it requires extension data to be a function, right?

Reply:
I think it's best to keep the extension data as is, in this case `unknown` is likely a bit better than `any` because it can avoid situations where type checking is skipped. The place where we actually do the type checking is `createComponentExtension` and from that point we assume that the data is sound

Reply:
Makes sense! Do you think I could address this in a separate pull request?

Reply:
https://github.com/backstage/backstage/pull/21652